drivers/HwWatchdog.h

```diff
@@ -0,0 +1,118 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_WATCHDOG_H
+#define MBED_WATCHDOG_H
+
+#ifdef DEVICE_WATCHDOG
+
+#include "watchdog_api.h"
+
+#include "mbed_error.h"
+
+#include "rtos/Mutex.h"
+
+#include "mbed.h"
+
+#include "LowPowerTimeout.h"
+
+#include <cstdio>
+
+namespace mbed {
+/** \addtogroup drivers */
+/** A system timer that will reset the system in the case of system failures or
+ *  malfunctions.
+ *
+ * Example:
+ * @code
+ *
+ * HwWatchdog watchdog;
+ * watchdog.start();
+ *
+ * while (true) {
+ *    wait(0.3);
+ *
+ * }
+ * @endcode
+ * @ingroup drivers
+ */
+class HwWatchdog {
+public:
+    HwWatchdog() {}
+
+public:
+    /** Start an independent watchdog timer
+     *
+     *
+     *  @return status MBED_SUCCESS if the watchdog timer was started
+     *                 successfully. MBED_ERROR_INVALID_ARGUMENT if one of the input
+     *                 parameters is out of range for the current platform.
+     *                 MBED_ERROR_INITIALIZATION_FAILED if one of the enabled input
+     *                 parameters is not supported by the current platform.
+     */
+    mbed_error_status_t start();
+
+
+    /** Refreshes the watchdog timer.
+     *
+     * This function should be called periodically before the watchdog times out.
+     * Otherwise, the system is reset.
+     *
+     * If the watchdog timer is not currently running this function does nothing
+     */
+    void kick();
+
+
+    /** Stops the watchdog timer
+     *
+     * Calling this function will attempt to disable any currently running
+     * watchdog timers if supported by the current platform.
+     *
+     * @return Returns MBED_SUCCESS if the watchdog timer was succesfully
+     *         stopped, or if the timer was never started. Returns
+     *         MBED_ERROR_OPERATION_PROHIBITED if the watchdog cannot be disabled
+     *         on the current platform.
+     */
+    mbed_error_status_t stop();
+
+
+    /** Get the watchdog timer refresh value
+     *
+     * This function returns the refresh timeout of the watchdog timer.
+     *
+     * @return Reload value for the watchdog timer in milliseconds.
+     */
+    uint32_t reload_value() const;
```

Comment:
When will user call this API? is it get API or set API? reload_value() - looks like user is reloading the value

Should it be - get_reload_value or get_refresh_value?

Reply:
This API required to user to read currently configured timeout value in  hw watchdog, this naming convention used earlier, I am just used it

Reply:
@c1728p9 @donatieng - reload_value - should it be get_reload_value?

Reply:
I'm not convinced. There's no harm in adding the `get` prefix, but it seems superfluous.